app/src/main/java/com/example/habitsmasher/ui/dashboard/DashboardFragment.java

```diff
@@ -74,7 +102,35 @@ public void onSwiped(@NonNull RecyclerView.ViewHolder viewHolder, int direction)
     };
 
     @Override
-    public void onDestroyView() {
-        super.onDestroyView();
+    public void addNewHabit(String title, String reason, Date date) {
```

Comment:
Good to see that all the database work has been moved to this method!

Now, we are missing some important **unit tests** for this method. 

Some example test scenarios include:
* Add habit is successful, check habit list and database got updated 
* Add habit is unsuccessful, check habit list did not get updated and database operation failed (some googling could be helpful here)
* Add habit with incorrect information/parameters passed in (if this is possible, I see we already do some filtering/validation in the UI, so if all the data flow into here is assumed to be **good**, then we don't need a test for this.